Mvc

Lexei4 opened the pull request.

Lexei4 committed:
Changes for AnnotationContext

Lexei4 committed:
Annotation fixes

Lexei4 committed:
Naming fix

Lexei4 committed:
Equipment added

Lexei4 committed:
ApplicationProperties added

Lexei4 committed:
Changes for MVC

Lexei4 committed:
MVC for Ingredient added

Lexei4 committed:
Handlers for POST requests  added

Lexei4 committed:
JSON Post examples added

Lexei4 committed:
GetRequest added

Lexei4 committed:
REST Controllers added

Lexei4 committed:
Method demonstaration added

Lexei4 committed:
Method demonstaration added

Lexei4 committed:
Documentation added

Lexei4 committed:
Documentation added

Lexei4 committed:
Main method demonstration added

Lexei4 committed:
JSON description added

Lexei4 committed:
Docs added

Lexei4 committed:
Bug fix

There was a bug implemented during cherry pick process.

Lexei4 committed:
Postman project added

Lexei4 committed:
BugFix #2

Primary anotation loss fixed

Lexei4 committed:
Task2 rework

Lexei4 committed:
Merge from previous homework

Lexei4 committed:
JavaDoc corrected

Lexei4 committed:
Javadoc#2

Lexei4 committed:
Set fields to private

ilya-sukhachev commented on build.gradle:
```diff
@@ -12,10 +14,18 @@ repositories {
-    )
+
+    implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
+    implementation('org.apache.logging.log4j:log4j-core:2.11.1')
```
используй единый стиль задания зависимостей.

ilya-sukhachev commented on build.gradle:
```diff
@@ -12,10 +14,18 @@ repositories {
+
+    implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
+    implementation('org.apache.logging.log4j:log4j-core:2.11.1')
     testCompile group: 'junit', name: 'junit', version: '4.12'
```
данная зависимость лишняя.

ilya-sukhachev replied:
testImplementation

ilya-sukhachev commented on src/main/java/ru/plushchov/config/SpringConfig.java:
```diff
@@ -0,0 +1,18 @@
+public class SpringConfig {
+
+        @Bean
+    public Executor threadPoolTaskExecutor() {
```
SpringConfig - зачем добавлен  ThreadPoolTaskExecutor это ДЗ_4.(замечание с прошлого ДЗ)

Lexei4 replied:
done

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/DeleteController.java:
```diff
@@ -0,0 +1,54 @@
+ */
+@RestController
+@RequestMapping("/api/delete")
+public class DeleteController {
```
контролеры делятся по сущности, а не по принципу действия.  типа тут все методы для удаления. переделать. Все комментарии по данному контролеру относятся и к остальным

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/DeleteController.java:
```diff
@@ -0,0 +1,54 @@
+
+    private DeleteService deleteService;
+
+    @Autowired
```
если 1 конструктор то Autowired ставить не надо

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/RegistrationController.java:
```diff
@@ -0,0 +1,115 @@
+
+
+
+//        ingredientDtoValidator.validate(ingredientDto, result);
```
лишние строчки и комментарии удаляем

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/RegistrationController.java:
```diff
@@ -0,0 +1,115 @@
+        }
+
+        registrationService.regIngredient(ingredientDto);
+        return new ResponseEntity<>(ingredientDto, HttpStatus.CREATED);
```
возвращай еще в хедере id  объекта показывал как это сделать на 4 лекции

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/RegistrationController.java:
```diff
@@ -0,0 +1,115 @@
+
+        if (result.hasErrors()) {
+            ingredientDto.setErrors(result.getAllErrors());
+            return new ResponseEntity<>(ingredientDto, HttpStatus.CONFLICT);
```
почему возвращаешь  HttpStatus.CONFLICT а не 400?

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/RegistrationController.java:
```diff
@@ -0,0 +1,115 @@
+    }
+
+    @PostMapping
+//    @RequestMapping(method = RequestMethod.POST)
```
комментарий зачем? убрать

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/RegistrationController.java:
```diff
@@ -0,0 +1,115 @@
+    }
+
+    @PostMapping
+//    @RequestMapping(method = RequestMethod.POST)
```
комментарий зачем? убрать

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/dto/BeverageDto.java:
```diff
@@ -0,0 +1,102 @@
+import java.util.Objects;
+import java.util.UUID;
+
+public class BeverageDto {
```
Присутствует документация: описаны все классы, методы интерфейсов, и public методы классов, в класы описаны все поля (JavaDoc). 
каждый класс должен иметь описание 
Аналогично и дальше по коду

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/examples/HelloWorldController.java:
```diff
@@ -0,0 +1,30 @@
+
+@Controller
+@RequestMapping("/api")
+public class HelloWorldController {
```
данный контролер был демонстраций определенных целей. Сейчас показываешь, что копируешь без думно. Удалить

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/examples/IndexController.java:
```diff
@@ -0,0 +1,15 @@
+@Controller
+public class IndexController {
+
+    @RequestMapping(value = "first", method = RequestMethod.GET)
```
данный контролер был демонстраций определенных целей. Сейчас показываешь, что копируешь без думно. Удалить

ilya-sukhachev commented on src/main/java/ru/plushchov/dao/BeverageDAO.java:
```diff
@@ -0,0 +1,29 @@
+     */
+    Beverage findBeverageById(UUID id);
+
+
```
лишние строчки

ilya-sukhachev commented on src/main/java/ru/plushchov/dao/BeverageDAOImpl.java:
```diff
@@ -0,0 +1,43 @@
+        return beverage;
+    }
+
+
```
лишние строчки. пиши чистый код. то есть похожий. между методами оставляй 1 пустую строчку не больше не меньше. Аналогично и во все остальном коде.

ilya-sukhachev commented on src/main/java/ru/plushchov/dao/RepositoryConfig.java:
```diff
@@ -0,0 +1,18 @@
+
+import org.springframework.context.annotation.Bean;
+
+public class RepositoryConfig {
```
зачем данный файл. используй конфигурирование над классами?

ilya-sukhachev commented on src/main/java/ru/plushchov/model/Ingredient.java:
```diff
@@ -0,0 +1,138 @@
+        return name;
+    }
+
+    public Ingredient() {
```
соблюдай порядок. поля, конструктор, методы. один стиль для всех классов.

ilya-sukhachev commented on src/main/java/ru/plushchov/service/BeverageServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    @Autowired
+    public void setBeverageDAO(BeverageDAO beverageDAO) {
```
у тебя есть конструктор зачем используешь на сеттером?

ilya-sukhachev commented on src/main/java/ru/plushchov/service/IngredientServiceImpl.java:
```diff
@@ -0,0 +1,54 @@
+        this.ingredientDAO = ingredientDAO;
+    }
+/**
+addIngridient is a Service method which will only work if price of the ingridient is less than one stipulated in
```
перед тем как отправлять код на ревью используй в каждом классе в  idea -> code -> ReformatCode
Либо галочка  при коммите в idea

ilya-sukhachev commented on src/main/java/ru/plushchov/service/REST/DeleteService.java:
```diff
@@ -0,0 +1,14 @@
+public interface DeleteService {
+    String deleteIngredient(UUID uuid);
+
+    String deleteBeverage(UUID uuid);
```
Присутствует документация: описаны все классы, методы интерфейсов, и public методы классов, в класы описаны все поля (JavaDoc). 
методы интерфейсов включая входные параметры и возвращаемые значения
Аналогично и по коду дальше

ilya-sukhachev commented on src/main/java/ru/plushchov/service/REST/DeleteService.java:
```diff
@@ -0,0 +1,14 @@
+/**
+ Интерфейс для обработки DELETE запросов DTO
+ * */
+public interface DeleteService {
```
делят по принадлежности сервиса а не принадлежности операции. у нас ООП

ilya-sukhachev commented on src/main/webapp/WEB-INF/dispatcher-servlet.xml:
```diff
@@ -0,0 +1,28 @@
+
+    <mvc:annotation-driven/>
+
+    <bean id="viewResolver" class="org.springframework.web.servlet.view.InternalResourceViewResolver">
```
это не нужно бы работает только с REST зачем тебе viewResolver?

Lexei4 committed:
Gradle dependencies refactor

Lexei4 committed:
Back to ClassPathXmlApplicationContext

Lexei4 committed:
Examples controllers deleted

скопировал, чтоы разобраться как работает

Lexei4 committed:
Javadoc updated

Lexei4 committed:
Codestyle review

Lexei4 committed:
Code cleaning

Lexei4 committed:
view-resolver deleted

Lexei4 committed:
Constructor moved

ilya-sukhachev commented on build.gradle:
```diff
@@ -12,10 +14,18 @@ repositories {
 
-    )
+
+    implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
```
используй один стиль задания зависимостей

Lexei4 replied:
В настоящий момент у меня в ветке это исправлено. Возможно github показывает не последние изменения. Сделаю новый pull-request, возможно поможет

ilya-sukhachev commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,5 +1,5 @@
 #Mon Oct 05 17:51:30 MSK 2020
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-bin.zip
```
с 5 задачи надо использовать 6.7

Lexei4 replied:
done

ilya-sukhachev commented on src/main/java/ru/plushchov/config/SpringConfig.java:
```diff
@@ -0,0 +1,18 @@
+public class SpringConfig {
+
+        @Bean
+    public Executor threadPoolTaskExecutor() {
```
это задача 4 удали

Lexei4 replied:
Аналогично, данный файл уже удален - сделаю новый pr

Lexei4 committed:
BeverageController implemented

Lexei4 committed:
IngredientController added

Lexei4 committed:
JavaDoc

Lexei4 committed:
EquipmentController added

Lexei4 committed:
EquipmentDAO bugfix

Lexei4 committed:
Unnecessary controllers deleted

Lexei4 committed:
POSTMAN collection updated

Lexei4 committed:
Added UUID to header

Lexei4 committed:
BeverageService refactoring for OOP

Lexei4 committed:
EquipmentService updated for OOP

Lexei4 committed:
IngredientController for OOP implemented

Lexei4 committed:
Refactoring - deleted unnecessary Services

Lexei4 committed:
Deleted unnecessary configs

Lexei4 committed:
POSTMAN collection updated

Lexei4 committed:
JavaDoc

Lexei4 committed:
Code reformat

в каждом классе в idea -> code -> ReformatCode

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/BeverageController.java:
```diff
@@ -0,0 +1,99 @@
+     */
+    private BeverageService beverageService;
+
+    @Autowired
```
если конструктор 1 не нужно писать Autowired. аналогично и дальше по коду

Lexei4 replied:
done

ilya-sukhachev commented on src/main/java/ru/plushchov/controller/BeverageController.java:
```diff
@@ -0,0 +1,99 @@
+     * @return
+     */
+    @GetMapping
+    public BeverageDto beverageRequestById(@RequestParam UUID id) {
```
какой RequestParam? @PathVariable используй. почитай статью в чате. Аналогично и в остальных урлах и в остальных контролерах

Lexei4 committed:
Gradle 6.7 implemented

Lexei4 committed:
Dependencies refactor

Изменения по рефакторингу были уже селаны - коммит для проверки корректного отображения в pull-request`е

Lexei4 committed:
Autowired annotation deleted

Lexei4 committed:
ParhVariable implemented
